src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2904,6 +2904,28 @@ internal override bool IsIOperationFeatureEnabled()
             return options?.IsFeatureEnabled(MessageID.IDS_FeatureIOperation) ?? false;
         }
 
+        public override string GetRequiredLanguageVersion(Diagnostic diagnostic)
+        {
+            if (diagnostic == null)
+            {
+                throw new ArgumentNullException(nameof(diagnostic));
+            }
+
+            bool found = false;
+            var foundVersion = LanguageVersion.Default;
+            foreach (var argument in diagnostic.Arguments)
+            {
+                if (argument is RequiredLanguageVersion versionDiagnostic)
+                {
+                    Debug.Assert(!found); // only one required language version in a given diagnostic
+                    found = true;
+                    foundVersion = versionDiagnostic.Version;
+                }
+            }
+
+            return found ? foundVersion.ToDisplayString() : null;
```

Comment:
I think `null` is better and consistent with other APIs. For instance,  `GetUnreferencedAssemblyIdentities` doesn't throw when using the wrong kind of diagnostic.